src/xrpld/rpc/handlers/LedgerEntry.cpp

```diff
@@ -774,7 +774,12 @@ doLedgerEntry(RPC::JsonContext& context)
         }
         else
         {
-            jvResult[jss::node] = sleNode->getJson(JsonOptions::none);
+            auto sleJson = sleNode->getJson(JsonOptions::none);
```

Comment:
Somehow this doesn't feel right. We did specialized parsing above and now we do it again.

Reply:
The specialized parsing above is only to find which object index given the request params. Here we are injecting the synthetic field after we've retrieved the object from DB. So I do believe this is the right place.

Reply:
But, I think we can refactor this. I think i can move this code to STLedgerEntry, let me refactor it

Reply:
Sure. But we have already parsed all ledger entries and we are of doing it again. What if more ledger entries require this special processing then we pretty much have to repeat specialized type parsing again. Maybe we could have a global json variable, which is set during the ledger entry parsing and then the values from this variable are copied into sleJson?